Automattic/jetpack
# AI Assistant: use AI Client suggestion hook. Clean the internal hook.

Janitorial/tidying PR. On this:

* Replace the usage of `useSuggestionsFromAI()` hook with the [useAiSuggestions()](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/ai-client/src/hooks/use-ai-suggestions) one that belongs to the AI Client kgg.
* Clean a typescript usage of the `useSuggestionsFromAI()` hook
* Remove completely the `code of the useSuggestionsFromAI()` 

Fixes https://github.com/Automattic/jetpack/issues/32076
Fixes https://github.com/Automattic/jetpack/issues/31936

## Proposed changes:

* AI Assistant: use the AI Client suggestion hook. Clean the internal hook.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Test all AI features
* AI Assistant block (this block keeps using uses the old useAISuggestionsFromOpenAI() hook, but still)
  * Create a block instance
  * Ask some request
  * Confirm the blocks provides generated text
* AI Extension
  * From en extended block (paragraph, heading...)
  * Request AI changes (translate, change the tone, correct grammar, etc)
  * Confirm the workflow works as expected
* Proofread 
  * Enable the feature
  * Open the Jetpack sidebar
  * Click on the AI Assistant panel
  * Click on the `Generate feedback` button
  * Confirm the  app opens the dialog and provides the expected feedback

They should generate context as usual.
